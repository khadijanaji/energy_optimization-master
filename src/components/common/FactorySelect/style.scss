@import "../../../style/colors.scss";
@import "../../../style/text.scss";

.factory-drawer {
    &__menu{
        position: relative;
        padding: 16px;
        box-shadow: 0 1px 1px 0 rgba(0,0,0,0.12);
    }
    &__select-text {
        color: rgba(0,0,0,0.87);
        font-family: 'Roboto';
        font-size: 14px;
        font-weight: 500;
        line-height: 24px;
        &__selected {
            position: relative;
        }
    }
    &__factory-selected {
        &__text {
            color: $text-color-selected;
            font-family: 'Roboto';
            font-size: 14px;
            font-weight: 500;
            line-height: 24px;
        }
        &__marker {
            position: absolute;
            width: 4px;
            background-color: $text-color-selected;
            left: 0;
            top: 0;
            height: 57px;
        }
    }
    &__icon {
        position: absolute;
        right: 16px;
        top: 16px;
        width: 30px;
        height: 30px;
        font-size: 16px;
    }
    &__item {
        display: inline-block;
        min-width: 180px;
        font-family: 'Roboto';
        padding-left: 32px;
        position: relative;
        color: black;

        &__selected {
            color: $text-color-selected;
        }
        &__marker {
            position: absolute;
            width: 4px;
            background-color: $text-color-selected;
            left: 0;
            height: 40px;
        }
    }
}

#factory-select-anchor {
    position: absolute;
    bottom: -33px;
}