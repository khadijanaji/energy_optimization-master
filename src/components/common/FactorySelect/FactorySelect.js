/* eslint-disable react/no-multi-comp */
/* eslint-disable react/jsx-handler-names */
/* eslint-disable react/no-set-state */
// @flow

import * as React from "react";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import {ArrowDropDown} from "@material-ui/icons";
import classNames from "classnames";
import "./style.scss";

type SelectState = {
  anchorEl: any
}
type SelectProps = {
  factories: Array<string>,
  currentEntity: string,
  currentSite: string
}

const isFactorySelected = (factory: string, currentEntity: string) => factory === currentEntity;

const renderFactorySelected = (currentEntity: string, clickHandler: Function) => (
  <div
    className="factory-drawer__factory-selected"
    onClick={clickHandler}
  >
    <div className="factory-drawer__factory-selected__marker" />
    <div className="factory-drawer__factory-selected__text">{currentEntity}</div>
  </div>
);

const renderNoFactorySelected = (clickHandler: Function) => (
  <div
    className="factory-drawer__select-text"
    onClick={clickHandler}
  >
    {"SÃ©lectionnez une usine"}
  </div>
);

class FactorySelect extends React.Component<SelectProps, SelectState> {
  constructor (props: SelectProps) {
    super(props);
    this.state = {"anchorEl": null};
    this.handleClick = this.handleClick.bind(this);
    this.handleClose = this.handleClose.bind(this);
  }

  handleClick: Function

  handleClick () {
    const anchor = document.getElementById("factory-select-anchor");
    this.setState({"anchorEl": anchor});
  }

  handleClose: Function

  handleClose () {
    this.setState({"anchorEl": null});
  }

  render () {
    const {anchorEl} = this.state;
    const {factories = [], currentEntity, currentSite} = this.props;
    return (
      <div className="factory-drawer__menu">
        {currentEntity && renderFactorySelected(currentEntity, this.handleClick)}
        {!currentEntity && renderNoFactorySelected(this.handleClick)}
        <div id="factory-select-anchor" />
        <div className="factory-drawer__icon">
          <ArrowDropDown onClick={this.handleClick} />
        </div>
        {factories && factories.length > 1 && <Menu
          anchorEl={anchorEl}
          id="factory-select-menu"
          onClose={this.handleClose}
          open={Boolean(anchorEl)}
        >
          {factories.map((factory) => (<MenuItem
            key={factory}
            onClick={this.handleClose}
          >
            {isFactorySelected(factory, currentEntity) && <div className="factory-drawer__item__marker" />}
            <a
              className={
                classNames(
                  "factory-drawer__item",
                  {"factory-drawer__item__selected": isFactorySelected(factory, currentEntity)}
                )}
              href={`#/${currentSite}/${factory}`}
            >
              {factory}</a>
          </MenuItem>))
          }
        </Menu>}
      </div>
    );
  }
}

export default FactorySelect;
