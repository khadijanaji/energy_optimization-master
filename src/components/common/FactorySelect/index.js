import FactorySelect from "./FactorySelect";
export default FactorySelect;
