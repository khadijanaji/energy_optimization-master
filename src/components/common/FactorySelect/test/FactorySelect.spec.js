/* globals describe it expect */

import {shallow} from "enzyme";
import React from "react";
import FactorySelect from "..";


describe("Component", () => {
  describe("FactoyList Render", () => {
    it("Without props: Should match snapshot", () => {
      const shallowRender = shallow(<FactorySelect />);
      expect(shallowRender).toMatchSnapshot();
    });

    it("With props: Should match snapshot", () => {
      const factories = [
        "JFC2",
        "JFC3",
        "JFC4"
      ];
      const shallowRender = shallow(<FactorySelect factories={factories} />);
      expect(shallowRender).toMatchSnapshot();
    });
  });
});
