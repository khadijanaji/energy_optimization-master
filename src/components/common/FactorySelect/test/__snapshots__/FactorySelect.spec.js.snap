// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FactoyList Render With props: Should match snapshot 1`] = `
<div
  className="factory-drawer__menu"
>
  <div
    className="factory-drawer__select-text"
    onClick={[Function]}
  >
    Sélectionnez une usine
  </div>
  <div
    id="factory-select-anchor"
  />
  <div
    className="factory-drawer__icon"
  >
    <pure(ArrowDropDown)
      onClick={[Function]}
    />
  </div>
  <WithStyles(Menu)
    anchorEl={null}
    id="factory-select-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      key="JFC2"
      onClick={[Function]}
    >
      <a
        className="factory-drawer__item"
        href="#/undefined/JFC2"
      >
        JFC2
      </a>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="JFC3"
      onClick={[Function]}
    >
      <a
        className="factory-drawer__item"
        href="#/undefined/JFC3"
      >
        JFC3
      </a>
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="JFC4"
      onClick={[Function]}
    >
      <a
        className="factory-drawer__item"
        href="#/undefined/JFC4"
      >
        JFC4
      </a>
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</div>
`;

exports[`Component FactoyList Render Without props: Should match snapshot 1`] = `
<div
  className="factory-drawer__menu"
>
  <div
    className="factory-drawer__select-text"
    onClick={[Function]}
  >
    Sélectionnez une usine
  </div>
  <div
    id="factory-select-anchor"
  />
  <div
    className="factory-drawer__icon"
  >
    <pure(ArrowDropDown)
      onClick={[Function]}
    />
  </div>
</div>
`;
